### PR TITLE
SOLR-16903: Update CLI tools to use java.nio.file.Path instead of java.io.File

### PR DESCRIPTION
**Pull request by AndreyBozhko**
https://issues.apache.org/jira/browse/SOLR-16903

# Description

I would like to work on SOLR-16903 - this PR only addresses part of it.

Here, I'm migrating the CLI tools from using `java.io.File` to `java.nio.file.Path` APIs.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/cli/PostToolTest.java [hunk 1]
```diff
@@ -281,15 +281,15 @@ public void testAppendUrlPath() {
   @Test
   public void testGuessType() {
     File f = new File("foo.doc");
```
Comment (epugh) [inline solr/core/src/test/org/apache/solr/cli/PostToolTest.java:283]: total nit pick, but would declaring the `File f` as a `Path p` save a `.toPath`???

**Review by epugh — APPROVED** [review]
LGTM.   One tiny nit pick...   I love to see these refactorings happening!

File: solr/core/src/test/org/apache/solr/cli/PostToolTest.java (hunk 1 above)
Comment (AndreyBozhko) [reply solr/core/src/test/org/apache/solr/cli/PostToolTest.java:283]: Yes - but I thought it was best to update the test files separately and at a later point.

Comment (epugh) [reply solr/core/src/test/org/apache/solr/cli/PostToolTest.java:283]: awesome!

**Comment by epugh** [comment]
@AndreyBozhko  let me know when you feel the PR is ready...!

**Comment by AndreyBozhko** [comment]
@epugh Thanks, I think this piece is ready.

**Comment by epugh** [comment]
Since this is purely a refactoring, skipping CHANGES.txt.  I am going to apply this to `main` only...

**Comment by epugh** [comment]
@AndreyBozhko please do tag me as you keep doing these!   I also like to keep an eye on https://ci-builds.apache.org/job/Solr/job/Solr-Test-main/ to make sure things remain good as these changes are made!

**Pull request merged**